gir-files/fix.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+set -x -e
+
+# Remove GLFuncs record
+# commit 5765641
+xmlstarlet ed --pf --inplace --delete '//_:record[@name="GLFuncs"]' GstGL-1.0.gir
+
+# Add a disguised GFuncs record (two steps)
+xmlstarlet ed --pf --inplace \
+	   --subnode '//_:namespace' --type elem -n 'recordTMP' --value ' ' \
+	   GstGL-1.0.gir
+
+xmlstarlet ed --pf --inplace \
+	   --insert '//_:recordTMP' -t attr -n 'name' --value 'GLFuncs' \
+	   --insert '//_:recordTMP' -t attr -n 'c:type' --value 'GstGLFuncs' \
+	   --insert '//_:recordTMP' -t attr -n 'disguised' --value '1' \
+	   --rename '//_:recordTMP' --value 'record' \
+	   GstGL-1.0.gir
+
+# incorrect GIR due bug #797144
+xmlstarlet ed --pf --inplace \
+	   --update '//*[@c:identifier="Dubois optimised Green-Magenta anaglyph"]/@c:identifier' \
+	     --value GST_GL_STEREO_DOWNMIX_ANAGLYPH_GREEN_MAGENTA_DUBOIS \
+	   --update '//*[@c:identifier="Dubois optimised Red-Cyan anaglyph"]/@c:identifier' \
+	      --value GST_GL_STEREO_DOWNMIX_ANAGLYPH_RED_CYAN_DUBOIS \
+	   --update '//*[@c:identifier="Dubois optimised Amber-Blue anaglyph"]/@c:identifier' \
+	      --value GST_GL_STEREO_DOWNMIX_ANAGLYPH_AMBER_BLUE_DUBOIS \
+	   GstGL-1.0.gir
+
+# remove constructors with native display
+xmlstarlet ed --pf --inplace \
+	   --delete '//_:constructor[@c:identifier="gst_gl_display_wayland_new_with_display"]' \
+	   --delete '//_:constructor[@c:identifier="gst_gl_display_x11_new_with_display"]' \
+	   --delete '//_:constructor[@c:identifier="gst_gl_display_egl_new_with_egl_display"]' \
+	   --delete '//_:function[@c:identifier="gst_gl_display_egl_get_from_native"]' \
+	   GstGL-1.0.gir
+
+# rename Display* to gpointer
+xmlstarlet ed --pf --inplace \
+	   --insert '//_:type[@c:type="Display*"]' -t attr -n 'name' --value 'gpointer' \
```

Comment:
This is X11 stuff and shouldn't be there at all. I'd suggest removing all X11/Wayland/EGL/etc specific API from the .gir file